apps/charterafrica/src/components/Comments/DynamicLineClampedTypography.js

```diff
@@ -0,0 +1,54 @@
+import { Button, useTheme } from "@mui/material";
+import PropTypes from "prop-types";
+import React, { forwardRef, useState, useRef } from "react";
+
+import LineClampedRichTypography from "@/charterafrica/components/LineClampedRichTypography";
+
+const DynamicLineClampedTypography = forwardRef(
+  ({ comment, ...props }, ref) => {
+    const ref1 = useRef();
+    const commentHeight = ref1?.current?.offsetHeight;
```

Comment:
`ref.current` is mutable and changing it  doesn't re-render the component so it would be pointless if executed useEffect.

PS. Even react's rules of hooks would flag it

Reply:
It's not about re-rendering; it's for static/server-side vs client-side rendering.

We want the `commentHeight` to be computed on the client the first time the component is mounted, hence the suggest to use useEffect.